eventbus.go

```diff
@@ -35,7 +35,17 @@ func (e EventBus) UnSubscribe(eventName string, ch DataChannel) {
 
 	for idx, subscriber := range subscribers {
 		if subscriber == ch {
-			e.events[eventName] = append(subscribers[:idx], subscribers[idx+1:]...)
+			// drain the channel
+			go func() {
+				defer close(ch)
+				for range ch {
+				}
+			}()
```

Comment:
```suggestion
			defer func() {
				close(ch)
				for range ch {
				}
			}()
```
I wonder can we do this?

Reply:
Your version is the correct one, mine was not closing it sometime. Apparently, after you close the channel the receive operations will return the zero value for the channel's type without blocking and this was confusing me.